consul/resource_consul_service_splitter_config_entry.go

```diff
@@ -0,0 +1,255 @@
+)
+
+const KindServiceSplitter = "service-splitter"
+
+var serviceSplitterConfigEntrySchema = map[string]*schema.Schema{
+	"name": {
+		Type:     schema.TypeString,
+		Required: true,
+		ForceNew: true,
+	},
+	"kind": {
+		Type:     schema.TypeString,
+		Required: false,
+		ForceNew: true,
+		Computed: true,
+	},
+	"partition": {
+		Type:     schema.TypeString,
+		Optional: true,
+		ForceNew: true,
+	},
+	"namespace": {
+		Type:     schema.TypeString,
+		Optional: true,
+		ForceNew: true,
+	},
+	"meta": {
+		Type:     schema.TypeMap,
+		Optional: true,
+		Elem:     &schema.Schema{Type: schema.TypeString},
+	},
+	"splits": {
+		Type:     schema.TypeList,
+		Required: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"weight": {
+					Type:     schema.TypeInt,
+					Required: true,
+				},
+				"service": {
+					Type:     schema.TypeString,
+					Required: true,
+				},
+				"service_subset": {
+					Type:     schema.TypeString,
+					Optional: true,
+				},
+				"namespace": {
+					Type:     schema.TypeString,
+					Optional: true,
+				},
+				"partition": {
+					Type:     schema.TypeString,
+					Optional: true,
+				},
+				"request_headers": {
+					Type:     schema.TypeSet,
+					Optional: true,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"add": {
+								Type:     schema.TypeMap,
+								Optional: true,
+								Elem:     &schema.Schema{Type: schema.TypeString},
+							},
+							"set": {
+								Type:     schema.TypeMap,
+								Optional: true,
+								Elem:     &schema.Schema{Type: schema.TypeString},
+							},
+							"remove": {
+								Type:     schema.TypeMap,
+								Optional: true,
+								Elem:     &schema.Schema{Type: schema.TypeString},
+							},
+						},
+					},
+				},
+				"response_headers": {
+					Type:     schema.TypeSet,
+					Optional: true,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"add": {
+								Type:     schema.TypeMap,
+								Optional: true,
+								Elem:     &schema.Schema{Type: schema.TypeString},
+							},
+							"set": {
+								Type:     schema.TypeMap,
+								Optional: true,
+								Elem:     &schema.Schema{Type: schema.TypeString},
+							},
+							"remove": {
+								Type:     schema.TypeMap,
+								Optional: true,
+								Elem:     &schema.Schema{Type: schema.TypeString},
+							},
+						},
+					},
+				},
+			},
+		},
+	},
+}
+
+func resourceServiceSplitterConfigEntry() *schema.Resource {
+
+	return &schema.Resource{
+		Create: resourceConsulServiceSplitterConfigEntryUpdate,
+		Update: resourceConsulServiceSplitterConfigEntryUpdate,
+		Read:   resourceConsulServiceSplitterConfigEntryRead,
+		Delete: resourceConsulServiceSplitterConfigEntryDelete,
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				parts := strings.Split(d.Id(), "/")
+				var kind, name, partition, namespace string
```

Comment:
We also don't need the kind during import because it will always be `service-splitter`.

Reply:
This is computed. So user will not be able to enter this.